imap_processing/codice/codice_l1a.py

```diff
@@ -80,101 +73,59 @@ class CoDICEL1aPipeline:
         self.table_id = table_id
         self.plan_id = plan_id
         self.plan_step = plan_step
         self.view_id = view_id
-        self.use_simulated_data = True
 
-    def _generate_simulated_data(self, length_in_bits):
-        """Return a list of random bytes to provide simulated science data.
-
-        This method is used as a temporary workaround to simulate science data
-        in the absence of real data to test with.
-
-        Parameters
-        ----------
-        length_in_bits : int
-            The number of bits used to generate the list of bytes. For example,
-            a ``length_in_bits`` of 80 yields a list of 10 bytes.
-
-        Returns
-        -------
-        random_bytes : bytes
-            A list of random bytes to be used as simulated science data
-        """
-        logger.info("\tUsing simulated data")
-        # Generate string of random bits of proper length
-        bit_string = bin(random.getrandbits(length_in_bits))[2:]
-        bit_string = bit_string.zfill(length_in_bits)
-        print(f"\tLength of data in bits: {len(bit_string)}")
-
-        # Convert list of random bits into byte-size list of integers
-        random_bytes_str = [bit_string[i : i + 8] for i in range(0, len(bit_string), 8)]
-        random_bytes_int = [int(item, 2) for item in random_bytes_str]
-        random_bytes = bytes(random_bytes_int)
-        print(f"\tLength of data in bytes: {len(random_bytes)}")
-
-        return random_bytes
-
-    def create_science_dataset(self):
+    def create_science_dataset(self) -> xr.Dataset:
         """Create an ``xarray`` dataset for the unpacked science data.
 
         The dataset can then be written to a CDF file.
 
         Returns
         -------
-        xarray.Dataset
+        xr.Dataset
             xarray dataset containing the science data and supporting metadata
 
         # TODO: Pull out common code and put in codice.utils alongside
         # create_hskp_dataset()
         """
-        # Define the attributes specific to species data
-        species_attrs = dataclasses.replace(
-            cdf_attrs.l1a_science_attrs,
-            depend_0="species",
-            depend_1=None,
-            depend_2=None,
-            catdesc="TBD",
-            fieldname="TBD",
-            label_axis="TBD",
-        ).output()
-
-        epoch_time = xr.DataArray(
-            [
-                launch_time for i in range(len(self.data))
-            ],  # Temporary workaround to get epoch in correct dimensions
+        epoch = xr.DataArray(
+            [calc_start_time(self.packets[0].data["SHCOARSE"].raw_value)],
             name="epoch",
             dims=["epoch"],
             attrs=ConstantCoordinates.EPOCH,
         )
 
-        species_array = xr.DataArray(
-            name="species",
-            data=np.zeros(36864),
-            dims=("species"),
-            attrs=species_attrs,
+        energy = xr.DataArray(
+            np.arange(128),
```

Comment:
Do we need to clarify that this is an index / steps in the name at all rather than just "energy"

Reply:
I will check with Joey on if the name matters within the CDF file, but for this python variable name, I can change it to `energy_steps`.